apitest/scripts/trade-simulation-utils.sh

```diff
@@ -221,6 +221,271 @@ getfirstofferid() {
+        printdate "BOB CLI: $CMD"
+        GETTRADE_CMD_OUTPUT=$(gettrade "$CMD")
+        exitoncommandalert $?
+        echo "$GETTRADE_CMD_OUTPUT"
+        printbreak
+
+        TRADE_DETAIL=$(gettradedetail "$GETTRADE_CMD_OUTPUT")
+        exitoncommandalert $?
+
+        IS_TRADE_DEPOSIT_CONFIRMED=$(istradedepositconfirmed "$TRADE_DETAIL" "TAKER")
+        exitoncommandalert $?
+        printdate "BOB $BOB_ROLE:  Has taker's trade deposit been confirmed?  $IS_TRADE_DEPOSIT_CONFIRMED"
+        printbreak
+
+        if [ "$IS_TRADE_DEPOSIT_CONFIRMED" = "YES" ]
+        then
+            DONE=1
+        else
+            printdate "Generating btc block while Bob waits for trade deposit to be confirmed."
+            genbtcblocks 1 0
+
+            RANDOM_WAIT=$(echo $[$RANDOM % 3 + 1])
+            sleeptraced "$RANDOM_WAIT"
+        fi
+    done
+}
+
+waitfortradepaymentsent() {
+    # Loops until buyer's trade payment has been sent.
+    PORT="$1"
+    SELLER="$2"
+    OFFER_ID="$3"
+    MAKER_OR_TAKER="$4"
+    DONE=0
+    while : ; do
+        if [ "$DONE" -ne 0 ]; then
+            break
+        fi
+
+        printdate "$SELLER:  Looking at trade with id $OFFER_ID."
+        CMD="$CLI_BASE --port=$PORT gettrade --trade-id=$OFFER_ID"
+        printdate "$SELLER CLI: $CMD"
+        GETTRADE_CMD_OUTPUT=$(gettrade "$CMD")
+        exitoncommandalert $?
+        echo "$GETTRADE_CMD_OUTPUT"
+        printbreak
+
+        TRADE_DETAIL=$(gettradedetail "$GETTRADE_CMD_OUTPUT")
+        exitoncommandalert $?
+
+        IS_TRADE_PAYMENT_SENT=$(istradepaymentsent "$TRADE_DETAIL" "$MAKER_OR_TAKER")
+        exitoncommandalert $?
+        printdate "$SELLER:  Has buyer's fiat payment been initiated?  $IS_TRADE_PAYMENT_SENT"
+        if [ "$IS_TRADE_PAYMENT_SENT" = "YES" ]
+        then
+            DONE=1
+        else
+            RANDOM_WAIT=$(echo $[$RANDOM % 3 + 1])
+            sleeptraced "$RANDOM_WAIT"
+        fi
+        printbreak
+    done
+}
+
+waitfortradepaymentreceived() {
+    # Loops until buyer's trade payment has been received.
+    PORT="$1"
+    SELLER="$2"
+    OFFER_ID="$3"
+    MAKER_OR_TAKER="$4"
+    DONE=0
+    while : ; do
+        if [ "$DONE" -ne 0 ]; then
+            break
+        fi
+
+        printdate "$SELLER:  Looking at trade with id $OFFER_ID."
+        CMD="$CLI_BASE --port=$PORT gettrade --trade-id=$OFFER_ID"
+        printdate "$SELLER CLI: $CMD"
+        GETTRADE_CMD_OUTPUT=$(gettrade "$CMD")
+        exitoncommandalert $?
+        echo "$GETTRADE_CMD_OUTPUT"
+        printbreak
+
+        TRADE_DETAIL=$(gettradedetail "$GETTRADE_CMD_OUTPUT")
+        exitoncommandalert $?
+
+        IS_TRADE_PAYMENT_SENT=$(istradepaymentreceived "$TRADE_DETAIL" "$MAKER_OR_TAKER")
+        exitoncommandalert $?
+        printdate "$SELLER:  Has buyer's payment been transferred to seller's fiat account?  $IS_TRADE_PAYMENT_SENT"
+        if [ "$IS_TRADE_PAYMENT_SENT" = "YES" ]
```

Comment:
Perhaps a comment to indicate that receipt is assumed as soon as payment sent message is received.

Reply:
Comment added in commit 072364b4902370fc1a3757640bec896917955f2d.